bench/tiered/push_pull.c

```diff
@@ -0,0 +1,259 @@
+ */
+static void
+run_test(const char *home, uint64_t num_records, bool flush)
+{
+    struct timeval start, end;
+
+    char buf[1024];
+    double diff_sec;
+    int64_t file_size;
+
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+
+    testutil_make_work_dir(home);
+    if (opts->tiered_storage) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/bucket", home));
+        testutil_make_work_dir(buf);
+    }
+
+    testutil_wiredtiger_open(opts, home, conn_config, NULL, &conn, false);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    /* Create and populate table. Checkpoint the data after that. */
+    testutil_check(session->create(session, opts->uri, table_config_row));
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), flush ? "flush_tier=(enabled,force=true)" : ""));
+
+    gettimeofday(&start, 0);
+    populate(session, num_records);
+    printf("Checkpoint buf : %s\n", buf);
+
+    testutil_check(session->checkpoint(session, buf));
+    gettimeofday(&end, 0);
+
+    diff_sec = difftime_sec(start, end);
+    printf("Code executed in %f ms, %f s\n", difftime_msec(start, end), diff_sec);
+
+    sleep(2);
+
+    get_file_size(home, &file_size);
+    testutil_assert(diff_sec > 0);
+    printf("File Size - %" PRIi64 ", Throughput - %f MB/second\n", file_size,
+      ((file_size / diff_sec) / MB));
+}
+
+/*
+ * populate --
+ *     TODO: Add a comment describing this function.
+ */
+static void
+populate(WT_SESSION *session, uint64_t num_records)
+{
+    WT_CURSOR *cursor;
+    WT_RAND_STATE rnd;
+    uint64_t i, str_len;
+
+    __wt_random_init_seed((WT_SESSION_IMPL *)session, &rnd);
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++)
+        data_str[i] = 'a' + (uint32_t)__wt_random(&rnd) % 26;
+
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+    for (i = 0; i < num_records; i++) {
+        cursor->set_key(cursor, i + 1);
+        cursor->set_value(cursor, data_str);
+        testutil_check(cursor->insert(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+    cursor = NULL;
+}
+
+/*
+ * get_file_size --
+ *     TODO: Retrieve the file size of the table.
+ */
+static void
+get_file_size(const char *home, int64_t *file_size)
+{
+    struct stat stats;
+
+    char path[512], pwd[512], token_path[512], stat_path[1512];
+    char *token;
+    const char *tablename;
+
+    size_t len;
+
+    token_path[0] = '\0';
+    tablename = strchr(opts->uri, ':');
+    testutil_assert(tablename != NULL);
+    tablename++;
+
+    if (getcwd(pwd, sizeof(pwd)) == NULL)
+        testutil_die(ENOENT, "No such directory");
+
+    testutil_check(__wt_snprintf(path, sizeof(path), "%s/%s", pwd, opts->argv0));
+
+    token = strtok(path, "/");
+
+    while (token != NULL) {
+        len = strlen(token);
+        strncat(token_path, "/", len);
+        strncat(token_path, token, len);
+
```

Comment:
I think we don't want the same logic that's in `testutil_deduce_build_dir`.  That is, I don't think it needs the tokenizing with strtok in a loop.  Instead, if we just use `home`, wouldn't it work?  Like:
```
           testutil_check(__wt_snprintf(stat_path, sizeof(stat_path), "%s/%s/%s-0000000001.wtobj",
              home, bucketname, tablename));
```
or
```
            testutil_check(__wt_snprintf(
              stat_path, sizeof(stat_path), "%s/%s.wt", home, tablename));
```
Am I missing something?

Reply:
You will need a loop to find all the files for the tiered case, like snprintf("%s/%s/%s-%10.10d.wtobj", home, bucketname, tablename, loop_index);

Reply:
Fixed unwanted tokenizing of the path to calculate the file size and added a function to iterate over all the tiered files and compute the file size.